boards/common/iotlab/include/periph_conf_common.h

```diff
@@ -149,19 +149,16 @@ static const uart_conf_t uart_config[] = {
 #define UART_NUMOF          ARRAY_SIZE(uart_config)
 /** @} */
 
+#if defined(MODULE_PERIPH_RTT) || defined(DOXYGEN)
```

Comment:
I think adding this preprocessor statement here is not needed. Can you explain you added it ?
If there's no good reason, then it should be removed.

Reply:
Because of this:

https://github.com/RIOT-OS/RIOT/blob/341916cd307bb9fcacf8a457f1d5988528c6ca5c/drivers/include/periph/rtt.h#L45-L56

Otherwise `RTT_FREQUENCY` would be defined but not `RTT_MAX_VALUE`... I could actually also remove the check and have those line do:

```
#ifndef RTT_FREQUENCY

/* Allow mock-RTT for unit tests */
#ifdef MOCK_RTT_FREQUENCY
#define RTT_FREQUENCY MOCK_RTT_FREQUENCY
#else

#warning "RTT_FREQUENCY undefined. Set RTT_FREQUENCY to the number of ticks" \
         "per second for the current architecture."
#endif
#endif

#ifndef RTT_MAX_VALUE 

/* Allow mock-RTT for unit tests */
#ifdef MOCK_RTT_FREQUENCY
#define RTT_MAX_VALUE MOCK_RTT_MAX_VALUE
#else

#warning "RTT_MAX_VALUE  undefined. Set RTT_MAX_VALUE  to the rtt timer size"
#endif
#endif
```

Reply:
This is rtt.h so you are supposed to have the `periph_rtt` module already added to the build. So this is IMHO a non issue :)
I would remove all preprocessor check (even in cpu): I tried locally and it just builds fine.

Reply:
> This is rtt.h so you are supposed to have the periph_rtt module already added to the build. So this is IMHO a non issue :)

 Not the case for `unittests`

Reply:
Thinking of it a second time. Your suggestion for rtt.h would make sense. Now I'm wondering if it is necessary to add this in this PR ?

Reply:
> Not the case for unittests

I could build and flash `tests-rtt-rtc` without issues.

Reply:
Using this diff:
```diff
diff --git a/boards/common/iotlab/include/periph_conf_common.h b/boards/common/iotlab/include/periph_conf_common.h
index b9bf38a95..948cf325a 100644
--- a/boards/common/iotlab/include/periph_conf_common.h
+++ b/boards/common/iotlab/include/periph_conf_common.h
@@ -149,7 +149,6 @@ static const uart_conf_t uart_config[] = {
 #define UART_NUMOF          ARRAY_SIZE(uart_config)
 /** @} */
 
-#if defined(MODULE_PERIPH_RTT) || defined(DOXYGEN)
 /**
  * @name    Real time counter configuration
  * @{
@@ -158,7 +157,6 @@ static const uart_conf_t uart_config[] = {
 #define RTT_FREQUENCY       (1)             /* in Hz */
 #endif
 /** @} */
-#endif /* defined(MODULE_PERIPH_RTT) || defined(DOXYGEN) */
 
 /**
  * @name    I2C configuration
diff --git a/boards/nucleo-f103rb/include/periph_conf.h b/boards/nucleo-f103rb/include/periph_conf.h
index 14fddd0df..4a142b5d0 100644
--- a/boards/nucleo-f103rb/include/periph_conf.h
+++ b/boards/nucleo-f103rb/include/periph_conf.h
@@ -119,14 +119,12 @@ static const uart_conf_t uart_config[] = {
 #define UART_NUMOF          ARRAY_SIZE(uart_config)
 /** @} */
 
-#if defined(MODULE_PERIPH_RTT) || defined(DOXYGEN)
 /**
  * @name    Real time counter configuration
  * @{
  */
 #ifndef RTT_FREQUENCY
 #define RTT_FREQUENCY       (16384)      /* in Hz */
-#endif
 /** @} */
 #endif /* defined(MODULE_PERIPH_RTT) || defined(DOXYGEN) */
 
diff --git a/cpu/stm32f1/include/periph_cpu.h b/cpu/stm32f1/include/periph_cpu.h
index fc6438495..1ec3f7480 100644
--- a/cpu/stm32f1/include/periph_cpu.h
+++ b/cpu/stm32f1/include/periph_cpu.h
@@ -133,7 +133,6 @@ typedef struct {
     uint8_t chan;           /**< CPU ADC channel connected to the pin */
 } adc_conf_t;
 
-#if defined(MODULE_PERIPH_RTT) || defined(DOXYGEN)
 /**
  * @name    Real time counter configuration
  * @{
@@ -150,7 +149,6 @@ typedef struct {
 /* RTC frequency of 32kHz is not recommended, see RM0008 Rev 20, p490 */
 #define RTT_MAX_FREQUENCY   (RTT_CLOCK_FREQUENCY / 2) /* in Hz */
 /** @} */
-#endif /* defined(MODULE_PERIPH_RTT) || defined(DOXYGEN) */
 
 #ifdef __cplusplus
 }
```

Reply:
Now try to build `bluepill`

Reply:
<details><summary>test-rtt-rtc</summary>

```
make IOTLAB_NODE=auto-ssh BOARD=iotlab-m3 -C tests/unittests tests-rtt_rtc flash test
make: Entering directory '/work/riot/RIOT/tests/unittests'
Building application "tests_unittests" for "iotlab-m3" with MCU "stm32f1".

"make" -C /work/riot/RIOT/boards/iotlab-m3
"make" -C /work/riot/RIOT/boards/common/iotlab
"make" -C /work/riot/RIOT/core
"make" -C /work/riot/RIOT/cpu/stm32f1
"make" -C /work/riot/RIOT/cpu/cortexm_common
"make" -C /work/riot/RIOT/cpu/cortexm_common/periph
"make" -C /work/riot/RIOT/cpu/stm32_common
"make" -C /work/riot/RIOT/cpu/stm32_common/periph
"make" -C /work/riot/RIOT/cpu/stm32f1/periph
"make" -C /work/riot/RIOT/drivers
"make" -C /work/riot/RIOT/drivers/periph_common
"make" -C /work/riot/RIOT/drivers/rtt_rtc
"make" -C /work/riot/RIOT/sys
"make" -C /work/riot/RIOT/sys/embunit
"make" -C /work/riot/RIOT/sys/isrpipe
"make" -C /work/riot/RIOT/sys/newlib_syscalls_default
"make" -C /work/riot/RIOT/sys/pm_layered
"make" -C /work/riot/RIOT/sys/stdio_uart
"make" -C /work/riot/RIOT/sys/test_utils/interactive_sync
"make" -C /work/riot/RIOT/sys/tsrb
"make" -C /work/riot/RIOT/tests/unittests/tests-rtt_rtc
   text	   data	    bss	    dec	    hex	filename
  13232	    132	   2700	  16064	   3ec0	/work/riot/RIOT/tests/unittests/bin/iotlab-m3/tests_unittests.elf
iotlab-node --jmespath='keys(@)[0]' --format='int'  --list saclay,m3,12 --flash /work/riot/RIOT/tests/unittests/bin/iotlab-m3/tests_unittests.bin | grep 0
0
r
ssh -t abadie@saclay.iot-lab.info 'socat - tcp:m3-12.saclay.iot-lab.info:20000'
READY
s
START
....
OK (4 tests)

make: Leaving directory '/work/riot/RIOT/tests/unittests'
```

</details>

Reply:
> Now try to build bluepill

Yes that's what I was testing :) and now I understand the issue: these boards don't provide the rtt feature and thus don't define RTT_FREQUENCY.

So I'm OK for modifying rtt.h like you suggested, this is the cleanest approach.

Reply:
> Now try to build bluepill

Thats why I added the first `if defined(MODULE_PERIPH_RTT)` in `periph_cpu.h`, which required the second for `iotlab-m3` and other :(

Reply:
> Thinking of it a second time. Your suggestion for rtt.h would make makes sense. Now I'm wondering if it is necessary to add this in this PR ?

Either Its done in a follow-up, and in that follow-up I remove the `defined(MODULE_PERIPH_RTT) || defined(DOXYGEN)`, or the other way around :)

Reply:
> Thinking of it a second time. Your suggestion for rtt.h would make makes sense. Now I'm wondering if it is necessary to add this in this PR ?

Did it in #13933, if that one gets in fast we can avoid the header guards :) (I would prefer that)

Reply:
#13933 is merged, please rebase and adapt this PR :)

Reply:
yep on it